### PR TITLE
Change the repo fetch script used in integration tests

### PR DESCRIPTION
**Pull request by nturgut**
change the repo fetch script to recognize candidate versions such as flutter-1.17-candidate.3. Originally the script only accepted branches such as v0.7.3 as valid engine branches.

This will fix test breakages such as: https://github.com/flutter/engine/pull/17931

### CODE REVIEW

File: tools/clone_flutter.sh [hunk 1]
```diff
@@ -19,9 +19,10 @@ cd $ENGINE_PATH/src/flutter
 # Special handling of release branches.
 ENGINE_BRANCH_NAME=`git branch | grep '*' | cut -d ' ' -f2`
 versionregex="^v[[:digit:]]+\."
+releasecandidateregex="^flutter-[[:digit:]]+\."
```
Comment (tvolkert) [inline tools/clone_flutter.sh:22]: Why not `^flutter-[[:digit:]]+\.[[:digit:]]+-candidate\.[[:digit:]]+$`?

Comment (nturgut) [reply tools/clone_flutter.sh:22]: makes sense. will do.

Comment (nturgut) [reply tools/clone_flutter.sh]: done.

**Comment by nturgut** [comment]
Thanks for the review. The tests passed, merging the PR.

**Pull request merged**